### PR TITLE
fixes #343: new data-source for ovirt_disk_attachments

### LINKED ISSUE #343: Implement `ovirt_vm_disk_attachments` data source
## Please describe what you would like to see

Implement a `ovirt_vm_disk_attachments` data source that returns all disk attachments for a VM.

## Please describe the solution you'd like

The data source should look like this:

```terraform
data "ovirt_vm_disk_attachments" "foo" {
  vm_id = var.vm_id
}
```

The result should contain the following fields:

-  `disk_id`

Optionally, it may contain all additional fields from the DiskAttachment objects from go-ovirt-client.

## Please describe your use case

We need to implement a resource that resizes all disks inherited from the template after VM creation. See openshift/installer#5867 for details.

### PR DESCRIPTION
**Pull request by engelmi**
## Please describe the change you are making

This PR fixes #343 and adds the ability to retrieve disk attachments by VM ID. 

## Are you the owner of the code you are sending in, or do you have permission of the owner?

yes

## The code will be published under the BSD 3 clause license. Have you read and understood this license?

yes

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically. ( keep in mind this list might be overwritten if the job XML is refreshed, for permanent whitelisting, please follow #3 option )
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

File: examples/data-sources/ovirt_disk_attachments/data.tf [hunk 1]
```diff
@@ -0,0 +1,10 @@
+data "ovirt_disk_attachments" "set" {
```
Comment (janosdebugs) [inline examples/data-sources/ovirt_disk_attachments/data.tf:1]: Please run `go generate` to generate the docs and make sure the docs make sense with this example.

**Review by janosdebugs — CHANGES_REQUESTED** [review]
Very good code, please run `go generate` and verify the docs.

File: ovirt/provider.go [hunk 2]
```diff
@@ -127,7 +127,9 @@ func (p *provider) getProvider() *schema.Provider {
 			"ovirt_nic":                      p.nicResource(),
 			"ovirt_tag":                      p.tagResource(),
 		},
-		DataSourcesMap: map[string]*schema.Resource{},
+		DataSourcesMap: map[string]*schema.Resource{
+			"ovirt_disk_attachments": p.diskAttachmentsDataSource(),
```
Comment (janosdebugs) [inline ovirt/provider.go:131]: This should read `ovirt_vm_disk_attachments`.

**Review by janosdebugs — CHANGES_REQUESTED** [review]

File: ovirt/provider.go (hunk 2 above)
Comment (engelmi) [reply ovirt/provider.go:131]: Yes, copy+paste error. I'll fix it.

**Review by janosdebugs — APPROVED** [review]

**Pull request merged**